### PR TITLE
feat(CX-3805): update add button behavior in my collection

### PR DESCRIPTION
**Pull request by MounirDhahri**
This PR resolves [] 

### Description

This PR updates the  Add button behavior in my collection to behave differently depending on where it was opened. The new behavior is as follows:
- If the screen was opened from the artists view -> Open **Add artist you collect** flow
- If the screen was opened from the artworks view -> Open **Add artwork to your collection** flow
- If the screen was opened from the default my collection view -> Open the **Default bottom sheet drawer** flow

https://github.com/artsy/eigen/assets/11945712/a0af02d3-f1d7-443e-a5f2-ee1a6eb15ebf

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- update add button behavior in my collection - mounir

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Review by lordkiz — DISMISSED** [review]
Cool.
But personally it is confusing to me if a button that previously showed only a bottomsheet sometimes doesn't and sometimes does, or just shows a form etc

**Review by olerichter00 — APPROVED** [review]
LGTM 🌟

File: src/app/Scenes/MyCollection/Components/MyCollectionStickyHeader.tsx [hunk 1]
```diff
@@ -92,13 +92,38 @@ export const MainStickyHeader: React.FC<{ hasArtworks: boolean }> = ({ hasArtwor
...
   const selectedTab = MyCollectionTabsStore.useStoreState((state) => state.selectedTab)
+
   const setSelectedTab = MyCollectionTabsStore.useStoreActions((actions) => actions.setSelectedTab)
   const setViewKind = MyCollectionTabsStore.useStoreActions((actions) => actions.setViewKind)
 
   const showAddToMyCollectionBottomSheet = debounce(() => {
     setViewKind({ viewKind: "Add" })
   }, 100)
 
+  const handlePlusButtonPress = () => {
```
[the remark this answers is not in the document]
Comment (MounirDhahri) [reply src/app/Scenes/MyCollection/Components/MyCollectionStickyHeader.tsx:103]: good comment - makes sense, I will update

**Comment by MounirDhahri** [comment]
@lordkiz - tbh, I agree 💯 with your comment, raised the same feedback but it's an intentional design decision.

**Pull request merged**